packages/pipeline-graph/src/render/render-node-content.tsx

```diff
@@ -11,28 +12,49 @@ export function RenderNodeContent(props: {
   node: AnyNodeInternal
   children?: React.ReactElement
   collapsed?: boolean
+  isFirst?: boolean
+  isLast?: boolean
+  parentNodeType?: ContainerNodeType
 }) {
-  const { node, children, collapsed } = props
+  const { node, children, collapsed, isFirst, isLast, parentNodeType } = props
   const { nodes } = useGraphContext()
 
   const nodeContent = nodes[node.type]
 
   switch (nodeContent.containerType) {
     case ContainerNode.leaf:
       return (
-        <nodeContent.component node={node as LeafNodeInternalType<{}>} collapsed={collapsed}>
+        <nodeContent.component
```

Comment:
not a blocker, but all react components should start with capital letters like `NodeContent.component`